.github/workflows/check-sdist.yml

```diff
@@ -0,0 +1,82 @@
+name: Check sdist
+
+# See: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency.
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+on:
+  push:
+    paths-ignore:
+      - "doc/**"
+      - "**/README.md"
+      - "CONTRIBUTING.md"
+    branches:
+      - master
+  pull_request:
+    paths-ignore:
+      - "doc/**"
+      - "**/README.md"
+      - "CONTRIBUTING.md"
+  release:
+    types:
+      - created
+
+jobs:
+
+  # Dogfood the generated sdist, to avoid bugs like https://github.com/haskell/cabal/issues/9833
+  # No caching, since the point is to verify they can be installed "from scratch"
+  # Don't run on master or a PR targeting master, because there's never an installable Cabal
+  dogfood-sdists:
+    name: Dogfood sdist on ${{ matrix.os }} ghc-${{ matrix.ghc }}
+    if: github.ref != 'refs/heads/master' && github.base_ref != 'master'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        # this should be kept up to date with the list in validate.yml, but should be the
+        # *first* compiler release so we validate against what is hopefully the first
+        # release of a corresponding Cabal and friends. it can also be short since it's
+        # highly unlikely that we are releasing really old branches.
+        ghc:
+          [
+            "9.10.1",
+            "9.8.1",
+            "9.6.1",
+            "9.4.1",
+          ]
```

Comment:
Does it buy us anything to try many versions? How about trying just one? I'm not saying it's the right way, just trying to understand..

Reply:
I'm trying to understand as well 😁 so this is a first cut and I expect to tune it later. I _think_ it'll be enough to include only ghcs that bundle `Cabal`s on release/maintained branches.